src/main/java/net/mcreator/ui/dialogs/imageeditor/FromTemplateDialog.java

```diff
@@ -507,17 +509,35 @@ private void generateFromTemplate(String template) {
 			cbs4.setSelectedItem(noimage);
 			break;
 		case "Ore":
-			cbs.setSelectedItem(ListUtils.getRandomItem(
-					templatesSorted.stream().filter(e -> e.toString().contains("noise")).collect(Collectors.toList())));
+			cbs.setSelectedItem(randomNoise);
 			cbs2.setSelectedItem(ListUtils.getRandomItem(
-					templatesSorted.stream().filter(e -> e.toString().matches("ore\\d+"))
-							.collect(Collectors.toList())));
+					templatesSorted.stream().filter(e -> e.toString().matches("ore\\d+")).toList()));
 			cbs3.setSelectedItem(noimage);
 			cbs4.setSelectedItem(noimage);
 			col1.setColor(ListUtils.getRandomItem(presetColors));
 			type1.setSelected(Math.random() < 0.4);
 			ang1.setValue(ListUtils.getRandomItem(new Integer[] { 0, 90, 180, 270, 0, 0 }));
 			break;
+		case "Block":
+			ResourcePointer randomTexture = ListUtils.getRandomItem(templatesSorted.stream()
+					.filter(e -> e.toString().contains("block") || e.toString().contains("pattern") || e.toString()
+							.contains("noise") || e.toString().contains("machine") || e.toString().contains("plank")
+							|| e.toString().contains("log") || e.toString().contains("wood") || e.toString()
+							.contains("leaves") || e.toString().contains("stone") || e.toString().contains("dirt"))
+					.toList());
+			cbs.setSelectedItem(randomTexture.toString().contains("noise") ?

```

Comment:
I can't because I'm using `randomTexture` in multiple places as well

Reply:
So this is to "marke sure" the texture for `cbs ` is not noise? Because you don't have a guarantee that the second time you call getRandomItem, you won't get noise this time

Reply:
I know, I just want to make sure that there's at least 1 block-like texture, so if `randomTexture` is a noise, then considering `randomTexture` is going to be `cbs2`, I set `cbs` to be completely random because, who cares?

Reply:
Then this check makes no sense, as it is not really doing what it should.

You could for example filter the list before making random selection to achieve this properly

Reply:
You want cbs to not be noise right? I don't see any other reason why you would write

`cbs.setSelectedItem(randomTexture.toString().contains("noise") ? ListUtils.getRandomItem(templatesSorted) : randomTexture);`

And if this is the case (you don't want noise), you can filter templatesSorted to not include noise when selecting random element for CBS (eg. call it `ResourcePointer randomTextureNotNoise).`

If this is not the case (you want noise), the just use randomTexture directly as

`cbs.setSelectedItem(randomTexture);`

Reply:
Because I want `cbs` and `cbs2` to be the same if `randomTexture` isn't a noise.

Reply:
That makes sense, but please then add a comment to the code explaining why code is written like so

Reply:
Done, this can be resolved